src/auction/src/orders.mo

```diff
@@ -464,8 +504,18 @@ module {
           ordersService.place(userInfo, chargeAcc, assetInfo, orderId, order);
           orderId;
         };
+        AssocList.replace<T.AssetId, ()>(assetIdSet, assetId, Nat.equal, ?()) |> (assetIdSet := _.0);
       };
 
+      List.toIter(assetIdSet)
+      |> Iter.map<(T.AssetId, ()), (T.AssetInfo, T.AssetId)>(_, func(aid, _) = (Vec.get(assets.assets, aid), aid))
+      |> (
+        switch (checkSessionNumberMismatch(_)) {
+          case (?aid) return #err(#placement({ error = #SessionNumberMismatch(aid); index = 0 }));
+          case (_) {};
+        }
+      );
```

Comment:
This whole block and the function checkSessionNumberMismatch can be replaced with a simple for loop:
```
for (aid in List.toIter(assetIdSet)) {
  ..
  // can return from the function here if mismatch
};
```

Reply:
done